src/App.tsx

```diff
@@ -1,24 +1,414 @@
+    const completedTodoIds = todo
+      .filter((element) => element.completed)
+      .map((element) => element.id);
+
+    try {
+      await Promise.all(
+        completedTodoIds.map((id) => client.delete(`/todos/${id}`)),
+      );
+
+      setTodo((prevTodo) => prevTodo.filter((element) => !element.completed));
+    } catch (error) {
+      console.log('There is an issue deleting completed todos.', error);
+      setDeleteErrorMessage('Unable to delete completed todos');
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, []);
+
+  useEffect(() => {
+    const isActive = todo.some((obj) => obj.completed === false);
+    const isFalse = todo.some((obj) => obj.completed === true);
+    const todoLength = todo.filter((obj) => {
+      return obj.completed === false;
+    });
+
+    setIsThereActiveTodo(isActive);
+    setIsThereCompletedTodos(isFalse);
+    setNumberOfActivTodos(todoLength.length);
+  }, [todo]);
+
+  const visibleTodos: Todo[] = useMemo(() => todo.filter((element) => {
+    switch (selectedTab) {
+      case SortType.Completed:
+        return element.completed;
+      case SortType.Active:
+        return !element.completed;
+      case SortType.All:
+        return todo;
+      default:
+        return todo;
+    }
+  }), [todo, selectedTab]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <>
+      <div className="todoapp">
+        <h1 className="todoapp__title">todos</h1>
+
+        <div className="todoapp__content">
+          <header className="todoapp__header">
+            <label htmlFor="nameInput">
+              <button
+                id="nameInput"
+                type="button"
+                className={classNames('todoapp__toggle-all', {
+                  active: isThereActiveTodo,
+                })}
+                onClick={changeAll}
+              >
+                {null}
+              </button>
+            </label>
+            <form onSubmit={handleFormSubmit}>
+              <input
+                type="text"
+                className="todoapp__new-todo"
+                placeholder="What needs to be done?"
+                value={inputValue}
+                onChange={(event) => {
+                  updatetempTodo(event.target.value);
+                }}
+              />
+            </form>
+          </header>
+          <section className="todoapp__main">
+            {todo.length !== 0 && (
+              <>
+                {visibleTodos.map((task) => {
+                  if (!isLoading) {
```

Comment:
Please use ternary operators inside jsx, it shortens your code and makes it more readable